package.json

```diff
@@ -0,0 +1,76 @@
+{
+  "name": "xterm-ligature-support",
+  "version": "1.0.0",
+  "description": "Add support for programming ligatures to xterm.js",
+  "homepage": "https://github.com/xtermjs/xterm-ligature-support#readme",
+  "repository": {
+    "type": "git",
+    "url": "https://github.com/xtermjs/xterm-ligature-support.git"
+  },
+  "bugs": {
+    "url": "https://github.com/xtermjs/xterm-ligature-support/issues"
+  },
+  "engines": {
+    "node": ">8.0.0"
+  },
+  "main": "lib/index.js",
+  "types": "lib/index.d.ts",
+  "scripts": {
+    "prepare": "node bin/download-fonts.js",
+    "lint": "tslint --project tsconfig.json --fix",
+    "prebuild": "npm run lint",
+    "build": "tsc",
+    "watch": "tsc -w",
+    "pretest": "npm run build",
+    "test": "nyc mocha lib/**/*.test.js"
+  },
+  "keywords": [
+    "font",
+    "ligature",
+    "xterm",
+    "xterm.js",
+    "terminal"
+  ],
+  "license": "MIT",
+  "dependencies": {
```

Comment:
From a dependency perspective, I'd bucket them into the following categories:

**Core functionality written for this package (hard to remove):**

 * font-finder
 * font-ligatures
 * promise-stream-reader
 * get-system-fonts

**Important to how the library works but parts may be inlineable:**

 * lru-cache
 * opentype.js
 * postcss-value-parser
 * lodash.clonedeep

**Unnecessary fluff:**

 * debug
 * fast-glob

What are the primary considerations w.r.t. dependencies? count, size, or depth? I wrote all of the packages in the first bucket so i can go over them with a fine-tooth comb to make sure nothing unnecessary creeps into the installed version (there's not a ton of code in any of them). The main one that adds a lot of weight is opentype.js, though a decent chunk of it is needed for handling the font file

Reply:
> What are the primary considerations w.r.t. dependencies? count, size, or depth?

All of the above, count/depth increases complexity and the likely hood of security issues, size increases the payload (it would be good to get opentype.js down some but luckily it's in JS so should be fairly small gzipped, not sure if it's worth the effort there).

Given your categorization I'd say we should look into the following:

- Think about sharing xterm.js' LRU map in the future (I think we were planning on simplifying this a lot when we drop some browser support)
- Inline postcss-value-parser, lodash.clonedeep
- Remove debug, fast-glob